modules/react/action-bar/lib/ActionBarItem.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import {composeHooks, EllipsisText, createSubcomponent} from '@workday/canvas-kit-react/common';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {PrimaryButton, SecondaryButton, PrimaryButtonProps} from '@workday/canvas-kit-react/button';
+
+import {useActionBarModel} from './useActionBarModel';
+import {useActionBarOverflow} from './useActionBarOverflow';
+import {useActionBarButtonProps} from './useActionBarButtonProps';
+
+export interface ActionBarItemProps extends PrimaryButtonProps {
+  /**
+   * Optionally pass index to action item. This should be done if `ActionBar.Item` components were created
+   * via a `Array::map` function. This index will ensure keyboard navigation works even if items are
+   * inserted out of order.
+   */
+  index?: number;
+  /**
+   * The contents of the action item. This will be the accessible name of the action for screen readers.
+   *
+   * ```tsx
+   * <ActionBar.Item>First Tab</ActionBar.Item>
+   * ```
+   */
+  children: React.ReactNode;
+  /**
+   * The identifier of the action. This identifier will be used for correct overflow behavior.
+   * If this property is not provided, it will default to a string representation
+   * of the the zero-based index of the Item when it was initialized.
+   */
+  'data-id'?: string;
+  /**
+   * Optional id. If not set, it will inherit the ID passed to the `ActionBar` component and append the
+   * index at the end. Only set this for advanced cases.
+   */
+  id?: string;
+  isPrimaryButton?: boolean;
```

Comment:
Is there a way for us to check what the first `item` is and automatically make it a `PrimaryButton` ?That way we don't even need this prop and the first item always passed is set to Primary, everything else is secondary? Maybe we can check model.state.items ? Again, thinking out loud, this is fine as it works!

Reply:
Model automatically makes the first item as a PrimaryButton. I'm not sure that it's reasonable to check items passed as children and change them to match a pattern "first item is a primary button". So it should be done manually as it's done for overflow functionality.